classes/property_list.php

```diff
@@ -347,4 +350,17 @@ private function array_serialize_cftypes(array $array) : array {
         $array = new CFDictionary($array); // Convert back to CFDictionary so serialization is recursive.
         return $array->toArray(); // Serialize.
     }
+
+    /**
+     * Check if an array is associative or sequential.
+     *
+     * @param array $array Array to check.
+     * @return bool False if not associative.
+     */
+    private function is_associative_array(array $array) {
+        if (array() === $array) {
+            return false;
+        }
+        return array_keys($array) !== range(0, count($array) - 1);
```

Comment:
as there is no unit tests to cover this complicated line. Can you please explain what's happening here?

Reply:
It checks that the array keys are 0,1,2,3...array_length - 1.